src/app/dataset-view/additional-components/metadata-component/components/set-transform/set-transform.component.ts

```diff
@@ -0,0 +1,332 @@
+import { TransformInput } from "./../../../../../api/kamu.graphql.interface";
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    OnInit,
+} from "@angular/core";
+import {
+    NgbModal,
+    NgbModalRef,
+    NgbTypeaheadSelectItemEvent,
+} from "@ng-bootstrap/ng-bootstrap";
+import { OperatorFunction, Observable } from "rxjs";
+import { debounceTime, distinctUntilChanged, switchMap } from "rxjs/operators";
+import { SearchApi } from "src/app/api/search.api";
+import AppValues from "src/app/common/app.values";
+import { DatasetAutocompleteItem } from "src/app/interface/search.interface";
+import { NestedTreeControl } from "@angular/cdk/tree";
+import { MatTreeNestedDataSource } from "@angular/material/tree";
+import { DatasetService } from "src/app/dataset-view/dataset.service";
+import { BaseComponent } from "src/app/common/base.component";
+import { MaybeNull } from "src/app/common/app.types";
+import { DatasetSchema } from "src/app/interface/dataset.interface";
+import {
+    EngineDesc,
+    EnginesQuery,
+    GetDatasetSchemaQuery,
+    SqlQueryStep,
+} from "src/app/api/kamu.graphql.interface";
+import { EngineService } from "src/app/services/engine.service";
+import * as monaco from "monaco-editor";
+import { EditSetTransformService } from "./edit-set-transform..service";
+import { DatasetInfo } from "src/app/interface/navigation.interface";
+import { ActivatedRoute, ParamMap } from "@angular/router";
+import { requireValue } from "src/app/common/app.helpers";
+import ProjectLinks from "src/app/project-links";
+import { DatasetHistoryUpdate } from "src/app/dataset-view/dataset.subscriptions.interface";
+import { DatasetNode, SetTransFormYamlType } from "./set-transform.types";
+import { ViewportScroller } from "@angular/common";
+import { FinalYamlModalComponent } from "../final-yaml-modal/final-yaml-modal.component";
+import { TemplatesYamlEventsService } from "src/app/services/templates-yaml-events.service";
+import { AppDatasetCreateService } from "src/app/dataset-create/dataset-create.service";
+@Component({
+    selector: "app-set-transform",
+    templateUrl: "./set-transform.component.html",
+    styleUrls: ["./set-transform.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SetTransformComponent extends BaseComponent implements OnInit {
+    public searchDataset = "";
+    private delayTime: number = AppValues.SHORT_DELAY_MS;
+    private inputDatasets = new Set([] as string[]);
+    public knownEngines: EngineDesc[];
```

Comment:
readonly?

Reply:
No, we assign the value in the subscription

Reply:
I think I've meant `delayTime`